app/src/main/java/org/mozilla/focus/permission/PermissionHandler.java

```diff
@@ -141,13 +146,18 @@ public void onRequestPermissionsResult(Context context, int requestCode, @NonNul
             return;
         }
         if (requestCode == actionId) {
-            setPermissionAsked(context, permissions[0]);
+
             if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                 permissionHandle.doActionGranted(permission, actionId, params);
                 clearAction();
             } else {
-                showSnackBar();
+                showToast();
```

Comment:
Tried if we don't reject, it's still incorrect. Maybe we should first show the dialog first and only if the user wants to give the site permission then we ask for device permission. This is how Firefox and Chrome handle this.